File: 5-Patron/python/LibroVentisca.py
import Libro


class LibroVentisca(Libro):
    def __init__(self, duracion, texto):
        super().__init__(duracion, texto)

    @property
    def duracion(self):
        return self.__duracion

    @duracion.setter
    def duracion(self, new):
        self.__duracion = new

    @property
    def texto(self):
        return self.__texto

    @texto.setter
    def duracion(self, new):
        self.__texto = new

    def verStats(self):
        return f"Libro ventisca \
                Duración: {self.__duracion} \
                Escrito antiguo: {self.__texto}"

File: 5-Patron/php/LibroVentisca.php
<?php
include './Libro.php';
class LibroVentisca extends Libro{
	private $duracion;
	private $texto;

	public function __construct($duracion, $texto){
    $this->duracion = $duracion;
    $this->texto = $texto;
	}

	public function getDuracion(){
		return $this->duracion;	
	}

	public function setDuracion($duracion){
		$this->duracion = $duracion;
	}

	public function getTexto(){
		return $this->texto;	
	}

	public function setTexto($texto){
		$this->texto = $texto;
	}

	public function verStats(){
    return  "Duración: " . $this->duracion .
            "Escrito antiguo: " . $this->texto;
  	}
}
?>
